pkg/controller/component/lifecycle/lfa_component.go

```diff
@@ -17,37 +17,42 @@ You should have received a copy of the GNU Affero General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-package handlers
+package lifecycle
 
 import (
 	"context"
 
-	"github.com/go-logr/logr"
-	"github.com/pkg/errors"
-	ctrl "sigs.k8s.io/controller-runtime"
-
-	"github.com/apecloud/kubeblocks/pkg/kb_agent/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type GRPCHandler struct {
-	Logger logr.Logger
+type postProvision struct {
+	namespace   string
+	clusterName string
+	compName    string
+}
+
+var _ lifecycleAction = &postProvision{}
+
+func (a *postProvision) name() string {
+	return "postProvision"
 }
 
-var _ Handler = &GRPCHandler{}
+func (a *postProvision) parameters(ctx context.Context, cli client.Reader) (map[string]string, error) {
+	return nil, nil
+}
+
+type preTerminate struct {
+	namespace   string
+	clusterName string
+	compName    string
+}
 
-func NewGRPCHandler(properties map[string]string) (*GRPCHandler, error) {
-	logger := ctrl.Log.WithName("GRPC handler")
-	h := &GRPCHandler{
-		Logger: logger,
-	}
+var _ lifecycleAction = &preTerminate{}
 
-	return h, nil
+func (a *preTerminate) name() string {
+	return "preTerminate"
 }
 
-func (h *GRPCHandler) Do(ctx context.Context, setting util.HandlerSpec, args map[string]any) (*Response, error) {
-	if setting.GPRC == nil {
-		return nil, errors.New("grpc setting is nil")
-	}
-	// TODO: implement grpc handler
-	return nil, ErrNotImplemented
+func (a *preTerminate) parameters(ctx context.Context, cli client.Reader) (map[string]string, error) {
+	return nil, nil
```

Comment:
and "$KB_CLUSTER_COMPONENT_IS_SCALING_IN"

in redis addon, it's used to determine whether rebalance should be performed when deleting a shard.